src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,408 @@
+                state.HPLInitialized = true;
+
+            }
+
+            this.HPLDirectory = this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, $"hpl-{this.Version}");
+            await this.stateManager.SaveStateAsync<HPLinpackState>($"{nameof(HPLinpackState)}", state, cancellationToken);
+
+            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, "Make.Linux_GCC"))
+                .ConfigureAwait(false);
+
+            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, "setup", "Make.Linux_GCC"))
+                .ConfigureAwait(false);
+
+            await this.ExecuteCommandAsync("bash", "-c \"source make_generic\"", this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup"), true, telemetryContext, cancellationToken);
+
+            await this.ConfigureMakeFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            await this.ExecuteCommandAsync("ln", $"-s {this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup", $"Make.Linux_GCC ")} Make.Linux_GCC", this.HPLDirectory, false, telemetryContext, cancellationToken);
+
+        }
+
+        /// <summary>
+        /// Executes the HPL workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            DateTime startTime = DateTime.UtcNow;
+            await this.ExecuteCommandAsync("make", $"arch=Linux_GCC", this.HPLDirectory, false, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.SetParameters();
+            await this.ConfigureDatFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            string results;
+            if (this.CpuArchitecture == Architecture.X64)
+            {
+                results = await this.ExecuteCommandAsync("runuser", $"-u azureuser -- mpirun --use-hwthread-cpus -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), true, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+            else
+            {
+                results = await this.ExecuteCommandAsync("runuser", $"-u azureuser -- mpirun -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), true, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+
+            DateTime endTime = DateTime.UtcNow;
+            this.LogMetrics(results, startTime, endTime, telemetryContext, cancellationToken);
+        }
+
+        private void SetParameters()
+        {
+            // gives you P*Q = Environment.ProcessorCount and P <= Q and  Q-P to be the minimum possible value.
+            this.ProcessRows = 1;
+            this.ProcessColumns = Environment.ProcessorCount;
```

Comment:
Get this processor count from systemmanager so that you could mock and test this